segtypes/common/codesubsegment.py

```diff
@@ -91,40 +95,48 @@ def process_insns(
         assert func_spim.vramEnd is not None
         self.parent: CommonSegCode = self.parent
 
-        func_sym = self.parent.create_symbol(func_spim.vram, type="func", define=True)
+        func_sym = self.get_most_parent().create_symbol(func_spim.vram, True, type="func", define=True)
         func_spim.setNameGetCallback(lambda _: func_sym.name)
+        if func_sym.rom is None:
+            func_sym.rom = func_spim.vromStart
 
         # Gather symbols found by spimdisasm and create those symbols in splat's side
         for referenced_vram in func_spim.referencedVRams:
             context_sym = self.text_section.getSymbol(
                 referenced_vram, tryPlusOffset=False
             )
             if context_sym is not None:
+                in_segment = context_sym.overlayCategory == self.get_most_parent().get_exclusive_ram_id()
```

Comment:
same here - I think I'm missing something, cause multiple segments can have the same exclusive_ram_id